option.go

```diff
@@ -142,6 +139,39 @@ func ZeroFields[T any](fields ...string) Option {
 	})
 }
 
+// KeepFields transforms a value of struct type T. It makes a copy of its input
+// preserving the specified field values and setting all other fields to their
+// zero values.
+//
+// This effectively makes comparison use only the provided fields.
+//
+// See also Transform.
+func KeepFields[T any](fields ...string) Option {
+	checkFieldsExist[T](fields...)
+	return Transform(func(v0 T) any {
+		var v1 T
+		e0 := reflect.ValueOf(&v0).Elem()
+		e1 := reflect.ValueOf(&v1).Elem()
+		for _, name := range fields {
+			if slices.Contains(fields, name) {
+				fv0 := e0.FieldByName(name)
+				fv1 := e1.FieldByName(name)
+				fv1.Set(fv0)
+			}
+		}
+		return v1
+	})
+}
+
+func checkFieldsExist[T any](fields ...string) {
```

Comment:
Nit: what do you think about []string here?

Reply:
Down with that. Will update.